pkg/jobrunaggregator/jobrunaggregatorapi/types_row_historical_data.go

```diff
@@ -72,6 +75,8 @@ func (a *AlertHistoricalDataRow) GetKey() string {
 		a.Platform,
 		a.Network,
 		a.Topology,
+		a.FirstObserved,
+		a.LastObserved,
```

Comment:
Do we want these in the keys?  LastObserved will be changing over time, if the keys in the maps that are generated change won't that break https://github.com/deepsm007/ci-tools/blob/33f9a92cb9fbd66e3759f084dbe7ef0f5074c57a/pkg/jobrunaggregator/jobrunhistoricaldataanalyzer/analyzer.go#L162

Have you tried running the update with an existing file and validated things still match up?